### ISSUE TITLE
Issue with IgnorePlugin

### ISSUE THREAD

**Issue by mrwillis** [mozilla-neutrino/neutrino-dev#830]
### Bug or issue?
Bug

Please try to answer the following questions:

- What version of Neutrino are you using? 8.2.3
- Are you trying to use any presets? If so, which ones, and what versions? neutrinojs/react: "^8.2.3",
- Are you using the Yarn client or the npm client? What version? yarn 1.6.0
- What version of Node.js are you using? 8.11.1
- What operating system are you using? Ubuntu 64
- What did you do? yarn run build (neutrino build)
- What did you expect to happen? It would produce files
- What actually happened, contrary to your expectations? I get this error when including the IgnorePlugin like so in my `.neutrino.js` file:
```js
    neutrino => neutrino.config.plugin('ignore')
      .use(webpack.IgnorePlugin, [/^\.\/locale$/, /moment$/]),
```

```
yarn run build
yarn run v1.6.0
$ neutrino build
⠋ Building project/home/julian/Desktop/cika/cika-client/node_modules/webpack/lib/IgnorePlugin.js:62
                compiler.hooks.normalModuleFactory.tap("IgnorePlugin", nmf => {
                               ^

TypeError: Cannot read property 'normalModuleFactory' of undefined
    at IgnorePlugin.apply (/home/julian/Desktop/cika/cika-client/node_modules/webpack/lib/IgnorePlugin.js:62:18)
    at Compiler.apply (/home/julian/Desktop/cika/cika-client/node_modules/neutrino/node_modules/tapable/lib/Tapable.js:375:16)
    at webpack (/home/julian/Desktop/cika/cika-client/node_modules/neutrino/node_modules/webpack/lib/webpack.js:33:19)
    at MapAction.ApAction$resolved$mapper [as mapper] (/home/julian/Desktop/cika/cika-client/node_modules/fluture/index.js:770:71)
    at MapAction$resolved [as resolved] (/home/julian/Desktop/cika/cika-client/node_modules/fluture/index.js:781:28)
    at resolved (/home/julian/Desktop/cika/cika-client/node_modules/fluture/index.js:220:19)
    at Resolved._fork (/home/julian/Desktop/cika/cika-client/node_modules/fluture/index.js:645:3)
    at drain (/home/julian/Desktop/cika/cika-client/node_modules/fluture/index.js:277:25)
    at settle (/home/julian/Desktop/cika/cika-client/node_modules/fluture/index.js:206:15)
    at resolved (/home/julian/Desktop/cika/cika-client/node_modules/fluture/index.js:220:5)
    at Immediate.EncaseP2$res [as _onImmediate] (/home/julian/Desktop/cika/cika-client/node_modules/fluture/index.js:1849:7)
    at runCallback (timers.js:796:20)
    at tryOnImmediate (timers.js:752:5)
    at processImmediate [as _immediateCallback] (timers.js:729:5)
error Command failed with exit code 1.
info Visit https://yarnpkg.com/en/docs/cli/run for documentation about this command.
```

### Feature request or enhancement?

Please describe your request in detail. Use the following questions as guidance: 

- What is the goal of the change?
- What are the pros and cons of the change?
- Could this dramatically improve the experience of our users?

**Comment by edmorley** [comment]
Hi! 

I created a test project using Neutrino's `create-project` and the React option, then:
* modified `.neutrinorc.js` to be:
```js
const webpack = require('webpack');

module.exports = {
  use: [
    '@neutrinojs/react',
    neutrino => neutrino.config.plugin('ignore')
      .use(webpack.IgnorePlugin, [/^\.\/locale$/, /moment$/]),
  ]
};
```
* ran `yarn add moment`
* added `import moment from 'moment';` to the top of `src/App.jsx`
* added `{moment().format()}` inside the `<h1>` in `src/App.jsx`

The project built successfully:
```
$ yarn build
yarn run v1.6.0
$ neutrino build
√ Building project completed
Hash: 0506667690228c9128c9
Version: webpack 3.11.0
Time: 11219ms
                                     Asset       Size        Chunks             Chunk Names
             index.cf2185d0daee76b9c4cc.js     165 kB         index  [emitted]  index
           runtime.a9edc78d5cd6127c9851.js  810 bytes       runtime  [emitted]  runtime
index.168567a0c7afd21a3ae09fc2aa649c84.css   18 bytes  index, index  [emitted]  index, index
                                index.html  503 bytes                [emitted]
Done in 16.31s.
```

...and opening index.html shows the current time.

Building again with the `IgnorePlugin` line commented out made `index.HASH.js` increase from 165KB to 466KB (so definitely worth doing!).

Looking at the stack trace in your comment I see what appears to be two different webpack directories being used?
`/home/julian/Desktop/cika/cika-client/node_modules/webpack/`
and
`/home/julian/Desktop/cika/cika-client/node_modules/neutrino/node_modules/webpack/`

Is it possible your `package.json` contains a `webpack` entry that is a different version from the one that is a dependency of the various Neutrino packages? Or is yarn package hoisting disabled in your environment?

@eliperelman if the above is the case, this is presumably another instance of #822 biting us.

**Comment by edmorley** [comment]
> Indeed my package.json contained a webpack 4! I started a fresh project and this problem went away. Perhaps there should be a warning about listing extra webpack entries?

Ah good to know that was it! I agree we should try and prevent this scenario - I think docs might be the only way - I've filed #849 to track that. Thankfully Neutrino v9 will come with webpack 4, so once it's released the problem will be harder to hit (at least until webpack 5 😆).

Closing this out since the original issue is now resolved and #849 open for the docs changes.
